client/src/page-not-found/fallback-link.tsx

```diff
@@ -32,8 +32,25 @@ export default function FallbackLink({ url }: { url: string }) {
     async (url) => {
       const response = await fetch(url);
       if (response.ok) {
-        const { doc } = await response.json();
-        return doc;
+        // If the URL is already for the JSON file, use  the response
+        if (response.url.endsWith("/index.json")) {
+          const { doc } = await response.json();
+          return doc;
+        }
+        // Otherwise, use the URL that gave the successful page (potentially
+        // including any redirects) and append index.json to get the data needed
+        var jsonURL = response.url;
+        if (!jsonURL.endsWith("/")) {
+          jsonURL += "/";
+        }
+        jsonURL += "index.json";
+        var jsonResponse = await fetch(jsonURL);
```

Comment:
Let's use `const` here:

```suggestion
        const jsonResponse = await fetch(jsonURL);
```